book/line_editor.md

```diff
@@ -1,124 +1,212 @@
 # Reedline, Nu's Line Editor
 
-Nushell's line editor [Reedline](https://github.com/nushell/reedline) is a
-cross-platform line reader designed to be modular and flexible. The engine is
-in charge of controlling the command history, validations, completions, hints
-and screen paint.
+Nushell's line-editor [Reedline](https://github.com/nushell/reedline) is
+cross-platform and designed to be modular and flexible. The line-editor is
+in charge of controlling the command history, validations, completions, hints,
+screen paint, and more.
 
-## Configuration
+[[toc]]
 
-### Editing Mode
+## Multi-line Editing
 
-Reedline allows you to edit text using two modes: vi and emacs. If not
-specified, the default edit mode is emacs mode. In order to select your
-favorite you need to modify your config file and write down your preferred
-mode.
+Reedline allows Nushell commandlines to extend across multiple lines. This can be accomplished using several methods:
 
-For example:
+1. Pressing <kbd>Enter</kbd> when a bracketed expression is open.
+
+   For example:
+
+   ```nu
+   def my-command [] {
+   ```
+
+   Pressing <kbd>Enter </kbd> after the open-bracket will insert a newline. This will also occur with opening (and valid) `(` and `[` expressions.
+
+   This is commonly used to create blocks and closures (as above), but also list, record, and table literals:
+
+   ```nu
+   let file = {
+     name: 'repos.sqlite'
+     hash: 'b939a3fa4ca011ca1aa3548420e78cee'
+     version: '1.4.2'
+   }
+   ```
+
+   It can even be used to continue a single command across multiple lines:
+
+   ```nu
+   (
+     ffmpeg
+     -i input.mp4
+     -vf "scale=1280:720,setsar=1:1"
+     -b:v 1500k
+     -preset veryfast
+     -crf 23
+     -c:a aac
+     -b:a 192k
+     -movflags +faststart
+     -y
+     output.mp4
+   )
+   ```
+
+2. Pressing <kbd>Enter</kbd> at the end of a line with a trailing pipe-symbol (`|`).
+
+   ```nu
+   ls                     |
+   where name =~ '^[0-9]' | # filenames starting with a digit
+   get name               | # get the filenames
+   mv ...$in ./backups/     # and move to backups folder
```

Comment:
I think the comments explaining the codes are unnecessary for the specific thing you want to explain. I think it would suffice to have one comment ` # comment after a trailing pipe are OK`

Reply:
Much better - Thanks!  I also left in a single "description" comment on the `where` in the next version just for consistency.